mitodl/mit-open
# Distinguish deps vs devDeps + make each package.json specify own dependencies

# What are the relevant tickets?
#257 

# Description (What does it do?)
This PR:
1. Moves development-only dependencies to `devDependencies`
2. Updates the package.json files so that each package specifies its own dependencies.  *This was a bit haphazard before*.
3. Enforces that each package.json specify its own dependencies moving forward using `[import/no-extraneous-dependencies`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-extraneous-dependencies.md#importno-extraneous-dependencies) ESLint rule. See [comment](https://github.com/mitodl/mit-open/pull/279/files#r1423257625) for more.
4. Adds `syncpack` to detect mismatched dependencies. See comment for more.

# Screenshots (if appropriate):


# How can this be tested?
1. Try importing a development dependency in a production file. (Example: `import { faker } from "@faker-js/faker/locale/en"` in HomePage.tsx. You should get a linting error.
2. Try importing a package not specified in a module's package.json; for example, `import { CKEditor } from "@ckeditor/ckeditor5-react"` in HomePage.tsx. You should get a linting error.
3. Visit http://localhost:8063; the app should work.

# Additional Context
